pontoon/db/__init__.py

```diff
@@ -30,6 +30,10 @@ def process_lhs(self, qn, connection):
         lhs, params = super(IContainsCollate, self).process_lhs(qn, connection)
         if self.collation:
             lhs = lhs.replace('::text', '::text COLLATE "{}"'.format(self.collation))
+            if '::text' not in lhs:
```

Comment:
@VishalCR7 Hey, I tested this locally and it works - a nice work Vishal!

The only thing I'm wondering if We should leave this hard-coded  "base.string" field. Currently, there's only one place that would be affected by this hack, but the whole goal of this class was to reuse this particular behavior in future components.

So, I experimented myself and got something like this:
https://gist.github.com/jotes/a3ee091c4f8d96a9d92d2067e110754d  This allows to handle basically any field in different models etc.
Anyway, I'm posting this only as an optional advice - feel free to express your opinion/ask questions.
Still, The current PR definitely solves the problem.

If that's okay, I'll leave the final decision/merge to @phlax.

Reply:
thanks for review @jotes - my thinking is that if we can reliably fix this for any field, then lets do that, however we do it.

Looking at posted gist, its probs more elegant, but i have an aversion to regexes unless absolutely necessary as they are hard to grok. I would be happy with either solution (if tests pass!)